test/src/com/xilinx/rapidwright/design/TestDesign.java

```diff
@@ -307,4 +308,20 @@ private static void testDesignWithBackslashHelper(Design design) {
         final int extraNets = 5; // {a, b, o, GLOBAL_USEDNET, GLOBAL_LOGIC0}
         Assertions.assertEquals(design.getNets().size(), 2 + extraNets);
     }
+    
+    @Test
+    public void testReplaceEDIFInDCP(@TempDir Path tempDir) {
+        Design design = RapidWrightDCP.loadDCP("picoblaze_ooc_X10Y235.dcp");
+        String unreadable = "picoblaze_ooc_X10Y235_unreadable_edif";
+        Path readableEDIF = tempDir.resolve(unreadable + ".edf");
+        design.getNetlist().exportEDIF(readableEDIF.toString());
+        Path unreadableDCP = RapidWrightDCP.getPath(unreadable + ".dcp");
+        Path readableDCP = tempDir.resolve("picoblaze_ooc_X10Y235.dcp");
+
+        ReplaceEDIFInDCP.main(new String[] {unreadableDCP.toString(), readableEDIF.toString(), readableDCP.toString()});
```

Comment:
Does this need a `Design.setAutoGenerateReadableEdif (false);` to ensure that if `ReplaceEDIFInDCP` were to fail then auto-generation doesn't kick in to save this test?